src/microbe_stage/Spawners.cs

```diff
@@ -342,24 +341,27 @@ public override IEnumerable<ISpawned> Spawn(Node worldNode, Vector3 location)
 /// <summary>
 ///   Spawns compound clouds of a certain type
 /// </summary>
-public class CompoundCloudSpawner : Spawner
+public class CloudSpawner : Spawner
```

Comment:
Couldn't the design work so that the CompoundCloudSpawner class is left as is, with the SpawnSystem handling the selecting which clouds are allowed to spawn?

Reply:
The old system had multipleCompoundCloudSpawners, one per compound. The new system has only one CloudSpawner, where the compounds are selected with the cloudBag. I figured this justified the name change.

Reply:
I did gather that from the changes. However, I think in order to decouple the spawn setup, from the actual spawning logic, you'll need to reintroduce back this intermediate CompoundCloudSpawner per type. Because as I said the spawn setup should *not* care at all what the player position is. The spawn system can have internally used a CloudSpawner that keeps track of extra state that is constructed from the registered CompoundCloudSpawner instances.

I'm being pretty hard on these changes, because the spawn system is a huge problem, and making it extra complicated with special logic for one type of spawning, might mean that it will be even more difficult to get someone to work on the spawn system again. Even with the current logic, which I think is pretty minimal for the random spawning it currently does, it's been years that I've tried to nudge people into working on the system. That's why I'd like to see a unified spawner design to avoid unnecessary complexity that will make future changes harder.

Though this issue mentions only clouds: https://github.com/Revolutionary-Games/Thrive/issues/1560 I think people would welcome having all the spawners moved over to a time based spawn algorithm that would still take the different densities into account.

Reply:
I understand why you are being hard on this, something like a spawn system should be easy to understand and change. I will rework it when I have free time.

Reply:
I look forward to that. It would be great to finally get the spawn system into better shape.
While pondering this a bit more, I think it could in fact simplify the design if the spawners are owned by the spawn system (and stored in saves, though getting delegates to save properly is not entirely trivial), and then the patch conditions could read the existing data, and communicate through method calls the necessary updates (as the spawn system probably needs to react to the changes, so it isn't a good idea to let the patch conditions applier to directly modify the spawner properties). But anyway, if you don't want to dig into the saving part yet, you can keep the current intermediate spawner objects, which as I said, make the behaviour regarding saving easier to *not* break.